src/sctools/bam.py

```diff
@@ -266,34 +407,11 @@ def split(
     if len(tags) == 0:
         raise ValueError('At least one tag must be passed')
 
-    def _cleanup(
-        _files_to_counts: Dict[pysam.AlignmentFile, int],
-        _files_to_names: Dict[pysam.AlignmentFile, str],
-        rm_all: bool = False,
-    ) -> None:
-        """Closes file handles and remove any empty files.
-
-        Parameters
-        ----------
-        _files_to_counts : dict
-            Dictionary of file objects to the number of reads each contains.
-        _files_to_names : dict
-            Dictionary of file objects to file handles.
-        rm_all : bool, optional
-            If True, indicates all files should be removed, regardless of count number, else only
-            empty files without counts are removed (default = False)
-
-        """
-        for bamfile, count in _files_to_counts.items():
-            # corner case: clean up files that were created, but didn't get data because
-            # n_cell < n_barcode
-            bamfile.close()
-            if count == 0 or rm_all:
-                os.remove(_files_to_names[bamfile])
-                del _files_to_names[bamfile]
+    if num_threads is None:
+        num_threads = multiprocessing.cpu_count()
 
     # find correct number of subfiles to spawn
-    bam_mb = os.path.getsize(in_bam) * 1e-6
+    bam_mb = sum(map(lambda in_bam: os.path.getsize(in_bam) * 1e-6, in_bams))
```

Comment:
generator comprehension is a bit easier to read here. 
```suggestion
    bam_mb = sum(os.path.getsize(b) * 1e-6 for b in in_bams)
```